torch/_inductor/mkldnn.py

```diff
@@ -525,7 +525,16 @@ def mkldnn_fuse_fx(gm: torch.fx.GraphModule, example_inputs):
     # the binary inputs have same tensor info(device, dtype, and layout).
 
     fake_mode = fake_mode_from_tensors(example_inputs)
-    ShapeProp(gm, fake_mode=fake_mode).propagate(*example_inputs)
+    # clone inputs to avoid side effects caused by inplace ops during propagate
+    tmp_example_inputs = list(
```

Comment:
it might not matter in mlkdnn but we should clone preserving storage aliasing/other properties. maybe through a util on FakeTensor